src/main/java/frc/robot/classes/TunableNumber.java

```diff
@@ -26,15 +31,18 @@ public TunableNumber(String name, double defaultValue, Consumer<Double> consumer
                 numberEntry = tuningTab.add(name, defaultValue).getEntry();
             }
         }
+        numberLog = new DoubleLogEntry(datalog, "/tunables/" + name); 
     }
 
     public void periodic(){
-        if (!Robot.isCompetition){
-            if (tuningMode){
-            value = numberEntry.getDouble(0);
-            consumer.accept(value);
+        if (tuningMode || !Robot.isCompetition){
+            double newValue = numberEntry.getDouble(0);
+            if (value != newValue){
+                consumer.accept(value);
```

Comment:
This isn't quite right.  You are assigning the old value, and not saving the newValue anywhere.